### PR TITLE
Add loading skeletons and an error message

### PR DESCRIPTION
**Pull request by ericatcorsha**
Before you submit your PR, make sure you check each of the following:

1. [x] I have read the [Contributing Guidelines](./CONTRIBUTING.md) and signed the [Contributor License Agreement](./CLA.md).
2. [x] My code is clean and ready-to-PR, including being free of lint errors and dead code
3. [x] I have **tested** my change / code and am confident that it works
4. [x] I have filled in the template below completely and accurately

##### Changelog

- add loading components for the analyses summary and analysis card

##### Testing

- verified that the new components look correct on the design system page
- observed that the loading state works correctly on the index page
- caused an error and observed that the error message appears correctly

### CODE REVIEW

File: ui/components/app/Analysis/core/index.tsx [hunk 1]
```diff
@@ -56,3 +56,27 @@ export const AnalysisCard: React.FC<AnalysisProps> = ({
...
   </Card>
 }
+
+export const AnalysisCardLoading = () => {
+  return (
+    <Card className="animate-pulse">
+      <CardContent>
+        <Typography variant='h2' className='h-6 bg-gray-200 rounded-full dark:bg-gray-500 w-48 mb-2'></Typography>
+        <Typography className='max-w-prose my-2 h-2.5 bg-gray-200 rounded-full dark:bg-gray-500 w-40 mb-2' variant='body1'></Typography>
+        <table className='font-light my-2 text-zinc-400 border-separate border-spacing-x-4'>
```
Comment (DavidCorsha) [inline ui/components/app/Analysis/core/index.tsx:66]: Are we using the table element in other places? I thought this was totally deprecated?

**Review by DavidCorsha — CHANGES_REQUESTED** [review]

File: ui/components/app/Analysis/core/index.tsx (hunk 1 above)
Comment (ericatcorsha) [reply ui/components/app/Analysis/core/index.tsx:66]: I copied the structure of the AnalysisCard defined in this file

File: ui/components/app/Analysis/core/index.tsx [hunk 2]
```diff
@@ -56,3 +56,27 @@ export const AnalysisCard: React.FC<AnalysisProps> = ({
...
+              <td><div className='max-w-prose my-2 h-4 bg-gray-200 rounded-full dark:bg-gray-500 w-20 mb-2'></div></td>
+              <td><div className='max-w-prose my-2 h-4 bg-gray-200 rounded-full dark:bg-gray-500 w-12 mb-2'></div></td>
+            </tr>
+          </tbody>
+        </table>
+        <div className='w-full bg-gray-200 dark:bg-gray-500 h-[400px] rounded-lg'> </div>
+      </CardContent>
+    </Card>
+  );
+}
```
Comment (JasonCorsha) [inline ui/components/app/Analysis/core/index.tsx:82]: i don't think we need the table structure, and some other minor details. could you try this out and see if you like it?

```
export const AnalysisCardLoading = () => {
  return (
    <Card className="animate-pulse">
      <CardContent className="space-y-3">
        <div className="flex items-center justify-between mb-2">
          <div className="flex space-x-2">
            <div className="w-8 h-8 bg-gray-200 rounded-full dark:bg-gray-500 " />
            <div className="h-8 bg-gray-200 rounded-full dark:bg-gray-500 w-64" />
          </div>
          <div className="flex">
            <div className="w-8 h-8 bg-gray-200 rounded-full dark:bg-gray-500 " />
          </div>
        </div>

        <span className="sr-only">Loading...</span>
        <div className="max-w-prose my-2 h-2.5 bg-gray-200 rounded-full dark:bg-gray-500 w-40 mb-2" />
        <div className="space-y-2.5 max-w-prose">
          <div className="flex items-center w-full space-x-2">
            <div className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-32" />
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24" />
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full" />
          </div>
          <div className="flex items-center w-full space-x-2 max-w-[480px]">
            <div className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-full" />
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full" />
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24" />
          </div>
          <div className="flex items-center w-full space-x-2 max-w-[400px]">
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full" />
            <div className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-80" />
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full" />
          </div>
          <div className="flex items-center w-full space-x-2 max-w-[480px]">
            <div className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-full" />
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full" />
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24" />
          </div>
          <div className="flex items-center w-full space-x-2 max-w-[440px]">
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-32" />
            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24" />
            <div className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-full" />
          </div>
        </div>
        <div className="w-full bg-gray-200 dark:bg-gray-700 h-[400px] rounded-lg"></div>
      </CardContent>
    </Card>
  );
};
```

File: ui/pages/index.tsx [hunk 3]
```diff
@@ -30,27 +25,32 @@ export default function Dashboard() {
...
             </div>
           </div>
         </main>
       </Layout>
     </>
   );
 }
+
+
+const AnalysisGridLoading = () => {
```
Comment (JasonCorsha) [inline ui/pages/index.tsx:48]: could you move this definition above line 12? it's an old javascript convention to put definitions above the lines that they're used in. the build process fixes it for us, but still

File: ui/components/app/Analysis/core/index.tsx (hunk 2 above)
Comment (ericatcorsha) [reply ui/components/app/Analysis/core/index.tsx]: done

File: ui/pages/index.tsx (hunk 3 above)
Comment (ericatcorsha) [reply ui/pages/index.tsx]: done

**Review by JasonCorsha — APPROVED** [review]
LGTM

**Review by DavidCorsha — APPROVED** [review]

**Pull request merged**